helix-term/src/commands/lsp.rs

```diff
@@ -108,6 +113,83 @@ fn sym_picker(
     .truncate_start(false)
 }
 
+fn diag_picker(
+    cx: &Context,
+    diagnostics: BTreeMap<lsp::Url, Vec<lsp::Diagnostic>>,
+    current_path: Option<lsp::Url>,
+    offset_encoding: OffsetEncoding,
+) -> FilePicker<(lsp::Url, lsp::Diagnostic)> {
+    // TODO: drop current_path comparison and instead use workspace: bool flag?
+
+    // flatten the map to a vec of (uri, diag) pairs
+    let mut flat_diag = Vec::new();
+    for (u, diags) in diagnostics {
+        for d in diags {
+            flat_diag.push((u.clone(), d));
+        }
+    }
+
+    let theme = cx.editor.theme.clone();
```

Comment:
This is a very pricy clone. Why not lookup the theme values ahead of time instead like we do in `ui/editor.rs`?